chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/state/AddedToRoom.kt

```diff
@@ -0,0 +1,33 @@
+package com.pusher.chatkit.rooms.state
+
+import com.pusher.chatkit.state.ChatkitState
+import org.reduxkotlin.reducerForActionType
+
+internal data class AddedToRoom(
+    val room: JoinedRoomInternalType,
+    val unreadCount: Int
+)
+
+internal val addedToRoomReducer =
+    reducerForActionType<ChatkitState, AddedToRoom> { state, action ->
+
+            val joinedRoom = Pair(action.room.id, action.room)
+            val joinedRoomUnreadCount = Pair(action.room.id, action.unreadCount)
+            var joinedRoomsState = JoinedRoomsState(
+                    hashMapOf(joinedRoom),
+                    hashMapOf(joinedRoomUnreadCount)
```

Comment:
let's abstract from concrete type, we don't really care here on implementation details of a map, what kotlin-stdlib provides as default `mapOf(...)` (indeed hash map + linked) is fine

Reply:
Gotcha. Have replaced all hashMapOfs with mapOf :)